_posts/2020-11-09-the-cloud-data-warehouse.md

```diff
@@ -0,0 +1,158 @@
+
+## Which Cloud Data Warehouse should we choose?
+
+In general I think it is more productive to match a Cloud Data Warehouse to your intended usage and workloads, than it is to compare Cloud Data Warehouse offerings against each other. Typical early considerations include
+* What cloud platforms (if any) you currently work with
+* Where your data is coming from
+* Volume of historic data and growth rate of data
+* Which geographic regions you operate in
+* How many users / workloads you will typically support
+
+However, here are some hot takes I'd make for general analytical workloads based in Australia
+
+I would favour or at least consider **BigQuery** when any of the following apply
+* Most of the data came from existing workloads in the same Google Cloud Region
+* Data Warehouse workload was focused on data science or machine learning
+* There is a business requirement to stream data to the warehouse in near realtime
+
+I would consider **RedShift** when any of the following apply
+* I have a fairly constant and stable utilization profile that is well suited to pre-purchasing reserved instances for cost optimization
+* I want to to provide a unified data warehouse access layer comprised of federated AWS sources (e.g. operational RDS databases, S3) in order to build out native RedShift data marts
+
+**For most other use cases, I would favour Snowflake**.
+
+You may notice that we haven't discussed Amazon Redshift too much in our consideration of Cloud Data Warehouse solutions; we have and continue to work with customers using RedShift and are happy to continue doing so. Yet there are reasons why we often reach for other solutions:
+
+* RedShift has struggled to keep pace with the feature sets of competitors such as BigQuery and Snowflake
+* It is generally more engineering intensive, considering administration, utilisation and optimization - managing infrastructure is still a customer concern
+* Storage and Compute separation is primitive - whilst this is improving with recent RA3 nodes, the price premium to be paid is a disincentive, for something that still doesn't match what competitors are offering (true on-demand usage)
+* Lack of high availability - RedShift clusters are limited to operating in a single availability zone. Whilst a cluster can generally recover simply from individual data node failures, a power loss to a whole AZ would require customers to create a new cluster from snapshots in a different AZ
+
+
+At the end of the day, it's what your organisation is capable of supporting that is more important, which is why I also rate this article on the [Two Philosophies of Cost in Data Analytics](https://www.holistics.io/blog/the-two-philosophies-of-cost-in-data-engineering/), discussing the trade off between service cost and labour cost.
```

Comment:
Clumsy. 

"At the end of the day, you need to choose a solution your organisation is capable of supporting. Just make sure you compare apples with apples however and consider both the labour cost and the service costs involved in your warehouse. For a great guide on this, refer to this article by...."